src-java/rule-manager/rule-manager-implementation/src/main/java/org/openkilda/rulemanager/factory/generator/flow/SingleTableIngressYRuleGenerator.java

```diff
@@ -60,14 +62,14 @@
         }
         result.add(command);
 
-        // TODO(tdurakov): since it's shared meter, this build might be moved outside.
-        SpeakerCommandData meterCommand = buildMeter(flowPath, config, sharedMeterId, sw);
-        if (meterCommand != null) {
-            addMeterToInstructions(sharedMeterId, sw, command.getInstructions());
-            result.add(meterCommand);
-            command.getDependsOn().add(meterCommand.getUuid());
+        if (generateMeterCommand) {
+            SpeakerCommandData meterCommand = buildMeter(externalMeterCommandUuid, flowPath, config, sharedMeterId, sw);
+            if (meterCommand != null) {
+                addMeterToInstructions(sharedMeterId, sw, command.getInstructions());
+                result.add(meterCommand);
+                command.getDependsOn().add(externalMeterCommandUuid);
+            }
```

Comment:
Looks like the second command in y-flow command pair (with `generateMeterCommand == false`) will not have correct dependency. Actually we need somehow check result of the first generator: are meter command available for the switch before setup dependency for the seccond command. Please add some unit tests on this scenario.

Reply:
Nice catch, fixed as we've discussed